src/actions/profile-view.js

```diff
@@ -446,9 +467,158 @@ function toggleOneTrack(
 }
 
 /**
- * This selects a track or several tracks from its reference and the passed
- * modifiers.
- * This will ultimately select the threads that this track belongs to, using its
+ * This computes the set of threads that's between two tracks (including
+ * themselves). This skips over hidden tracks.
+ */
+function findThreadsBetweenTracks(
+  state: State,
+  fromTrack: TrackInformation,
+  toTrack: TrackInformation
+): Array<ThreadIndex> {
+  const globalTracks = getGlobalTracks(state);
+  const globalTrackOrder = getGlobalTrackOrder(state);
+  const hiddenGlobalTracks = getHiddenGlobalTracks(state);
+
+  const foundThreadIndexes = [];
+
+  // Where are they located in the track order?
+  const fromGlobalOrder = globalTrackOrder.indexOf(fromTrack.globalTrackIndex);
+  const toGlobalOrder = globalTrackOrder.indexOf(toTrack.globalTrackIndex);
+
+  // TODO Check the relative order of from and to, and possibly invert them
+
+  for (
+    let globalOrderIndex = fromGlobalOrder;
+    globalOrderIndex <= toGlobalOrder;
+    globalOrderIndex++
+  ) {
+    const globalTrackIndex = globalTrackOrder[globalOrderIndex];
+    if (hiddenGlobalTracks.has(globalTrackIndex)) {
+      continue;
+    }
+    const globalTrack = globalTracks[globalTrackIndex];
+    if (globalTrack.type !== 'process') {
+      continue;
+    }
+
+    const localTrackOrder = getLocalTrackOrder(state, globalTrack.pid);
+    const hiddenLocalTracks = getHiddenLocalTracks(state, globalTrack.pid);
+    const localTracks = getLocalTracks(state, globalTrack.pid);
+
+    let shouldAddStartGlobalTrack = true;
+    let localTrackOrderStart = 0;
+    let localTrackOrderEnd = localTrackOrder.length - 1;
+    if (globalOrderIndex === fromGlobalOrder) {
+      // This is the first process group, this means we possibly shouldn't add
+      // all tracks.
+      if (fromTrack.type === 'local') {
+        shouldAddStartGlobalTrack = false;
+        localTrackOrderStart = localTrackOrder.indexOf(
+          fromTrack.localTrackIndex
+        );
+      }
+    }
+
+    if (globalOrderIndex === toGlobalOrder) {
+      // This is the last process group, this means we possibly shouldn't add
+      // all tracks.
+      if (toTrack.type === 'global') {
+        // No local track should be added
+        localTrackOrderEnd = -1;
+      } else {
+        localTrackOrderEnd = localTrackOrder.indexOf(toTrack.localTrackIndex);
+      }
+    }
+
+    // Add the global track if it's not a virtual track and not out of the range.
```

Comment:
By "virtual" you mean a global tack with a thread, right?

Reply:
"virtual track" means here a global track that is _not_ a thread (such as what we get when importing sometimes).

But the comment says "add the global track if it's _not_ virtual", maybe that's where you got confused?